src/ObjectUtils/PdbFileLLVM.h

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef OBJECT_UTILS_PDB_FILE_LLVM_H_
+#define OBJECT_UTILS_PDB_FILE_LLVM_H_
+
+#include <llvm/DebugInfo/CodeView/CVSymbolVisitor.h>
+#include <llvm/DebugInfo/CodeView/CodeView.h>
+#include <llvm/DebugInfo/CodeView/GUID.h>
+#include <llvm/DebugInfo/CodeView/SymbolDeserializer.h>
+#include <llvm/DebugInfo/CodeView/SymbolVisitorCallbackPipeline.h>
+#include <llvm/DebugInfo/CodeView/SymbolVisitorCallbacks.h>
+#include <llvm/DebugInfo/MSF/MappedBlockStream.h>
+#include <llvm/DebugInfo/PDB/Native/DbiStream.h>
+#include <llvm/DebugInfo/PDB/Native/ModuleDebugStream.h>
+#include <llvm/DebugInfo/PDB/Native/NativeSession.h>
+#include <llvm/DebugInfo/PDB/Native/PDBFile.h>
+#include <llvm/DebugInfo/PDB/PDB.h>
+#include <llvm/DebugInfo/PDB/PDBSymbolExe.h>
+#include <llvm/DebugInfo/PDB/PDBTypes.h>
+#include <llvm/Demangle/Demangle.h>
+
+#include <array>
+#include <filesystem>
+
+#include "ObjectUtils/PdbFile.h"
+#include "ObjectUtils/WindowsBuildIdUtils.h"
+#include "OrbitBase/Result.h"
+#include "symbol.pb.h"
+
+namespace orbit_object_utils {
+
+class PdbFileLLVM : public PdbFile {
+ public:
+  PdbFileLLVM(std::filesystem::path file_path, const ObjectFileInfo& object_file_info,
+              std::unique_ptr<llvm::pdb::IPDBSession> session);
+
+  [[nodiscard]] ErrorMessageOr<orbit_grpc_protos::ModuleSymbols> LoadDebugSymbols() override;
+  [[nodiscard]] const std::filesystem::path& GetFilePath() const override { return file_path_; }
+
+  [[nodiscard]] std::array<uint8_t, 16> GetGuid() const override {
```

Comment:
What about using `sizeof(llvm::codeview::GUID::Guid)` here instead of the hard-coded 16? This means we could also avoid the kGuidSize constant below and we would also ensure that the Guid always fits in the output buffer.

If you prefer to keep it as-is, I would at least add a static_assert that ensures that `guid.Guid` isn't larger than the output buffer.

Reply:
Thanks, I added a static assert, GetGuid() is pure virtual and we wouldn't want the dependency on llvm in the interface...

Reply:
I see. That makes sense. Thank you!